src/app/business/delfin/storages/storages.component.ts

```diff
@@ -592,16 +656,29 @@ export class StoragesComponent implements OnInit {
             console.log("Something went wrong. Could not initiate the sync.", error);
         });
     }
-    
-
-    
-
-    
-
-    lazyLoadTreeChildren(event){
 
+   getVolumesByStorage(storageId){
+        // Get all the volumes associated with the Storage device
+        this.ds.getAllVolumes(storageId).subscribe((res)=>{
+            let vols = res.json().volumes;
+           return vols;
+        }, (error)=>{
+            console.log("Something went wrong. Could not fetch Volumes.", error)
+        });
+       
     }
 
+    getPoolsByStorage(storageId){
```

Comment:
These changes are not for this PR, right?

Reply:
These are helper functions that I have created .
Not for this feature but part of the enhancement.